packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -131,16 +136,38 @@ export default async (req, res) => {
     regionId = regions[0].id
   }
 
+  const toCreate: {
+    region_id: string
+    context: object
+    sales_channel_id?: string
+    customer_id?: string
+    email?: string
+    shipping_address?: Partial<AddressPayload>
+  } = {
+    region_id: regionId,
+    sales_channel_id: validated.sales_channel_id,
+    context: {
+      ...reqContext,
+      ...validated.context,
+    },
+  }
+
+  if (req.user && req.user.customer_id) {
+    const customerService = req.scope.resolve("customerService")
+    const customer = await customerService.retrieve(req.user.customer_id)
+    toCreate["customer_id"] = customer.id
+    toCreate["email"] = customer.email
+  }
+
+  if (validated.country_code) {
+    toCreate["shipping_address"] = {
+      country_code: validated.country_code.toLowerCase(),
+    }
```

Comment:
todo: In case the validated data contains a `country_code`, we are overwriting `cart.shipping_address` entirely.

We should either spread the address:
```
 toCreate["shipping_address"] = {
      ... toCreate["shipping_address"],
      country_code: validated.country_code.toLowerCase(),
    }
```
or assign it directly:

```
 toCreate["shipping_address"]["country_code"] = validated.country_code.toLowerCase()
```

And we should gracefully handle address being `undefined` for both cases.

Reply:
`shipping_address` cannot currently be set using the `POST /carts` so not a concern in this PR